packages/react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -140,6 +142,7 @@ export function getResponsiveMode(currentWindow: Window | undefined): Responsive
     } catch (e) {
       // Return a best effort result in cases where we're in the browser but it throws on getting innerWidth.
       responsiveMode = getInitialResponsiveMode();
+      throw new Error('Failed to get innerWidth on the current window, falling back to default responisve mode');
     }
```

Comment:
Rethrowing the error will cause React to de-render all content. I think that's a bit much for simply failing to get the responsive breakpoint.

Reply:
Agree, checking if we can stop this error from occurring at the first place